scripts/generateCliCode.ts

```diff
@@ -0,0 +1,340 @@
+ * in `assets/specs/command-api.yaml`
+ *
+ * "flagType": capitalized name of the type given in command-api.yaml. This maps to code like
+ *             `cmdFlags.StringVar(...)` in the generated go code
+ */
+
+interface commandFlagType {
+  capitalizedName: string;
+  defaultValue: string;
+  flagOption: string;
+  flagType: goCmdFlagTypeName;
+  lcTypeName: goTypeName;
+  propertyName: string;
+  aliasFor: string;
+  enums: string;
+  usageNote: string;
+  valuePart: string;
+}
+
+type yamlObject = any;
+
+type goTypeName = 'string' | 'bool' | 'int';
+type goCmdFlagTypeName = 'String' | 'Bool' | 'Int';
+type typeValue = goTypeName | settingsTreeType;
+type settingsTypeObject = { type: typeValue };
+type settingsTreeType = Record<string, settingsTypeObject>;
+
+function assert(predicate: boolean, error: string) {
+  if (!predicate) {
+    throw new Error(error);
+  }
+}
+
+function capitalize(s: string) {
+  return s[0].toUpperCase() + s.substring(1);
+}
+
+function capitalizeParts(s: string) {
+  return s.split('.').map(capitalize).join('.');
+}
+
+function lastName(s: string): string {
+  return s.split('.').pop() ?? '';
+}
+
+class Generator {
+  constructor() {
+    this.commandFlags = [];
+    this.settingsTree = {};
+  }
+
+  commandFlags: Array<commandFlagType>;
+  settingsTree: settingsTreeType;
+
+  protected async loadInput(inputFile: string): Promise<yamlObject> {
+    const contents = (await fs.promises.readFile(inputFile)).toString();
+
+    try {
+      return yaml.parse(contents);
+    } catch (e) {
+      console.error(`Can't parse input file ${ inputFile }\n${ contents }\n\nError: ${ e }`, e);
+      throw (e);
+    }
+  }
+
+  protected processInput(obj: yamlObject, inputFile: string): void {
+    const preferences = obj?.components?.schemas?.preferences;
+
+    if (!preferences) {
+      throw new Error(`Can't find components.schemas.preferences in ${ inputFile }`);
+    }
+    assert(preferences.type === 'object', `Expected preferences.type = 'object', got ${ preferences.type }`);
+    assert(Object.keys(preferences.properties).length > 0, `Not a properties object: ${ preferences.properties }`);
+    for (const propertyName of Object.keys(preferences.properties)) {
+      this.walkProperty(propertyName, preferences.properties[propertyName], this.settingsTree);
+    }
+  }
+
+  protected async emitOutput(outputFile: string) {
+    const options = { rmWhitespace: false };
+    const templateFile = 'scripts/assets/options.go.templ';
+
+    const linesForJSON = this.collectServerSettingsForJSON(this.settingsTree, true, '');
+    const linesWithoutJSON = this.collectServerSettingsForJSON(this.settingsTree, false, '');
+    const data = {
+      commandFlags:     this.commandFlags,
+      linesForJSON:     linesForJSON.join('\n'),
+      linesWithoutJSON: linesWithoutJSON.join('\n'),
+    };
+    const renderedContent = await ejs.renderFile(templateFile, data, options).then((rc: string) => rc);
```

Comment:
```suggestion
    const renderedContent = await ejs.renderFile(templateFile, data, options);
```
It's possible that the last function (which appears to be identity) is doing something useful, but I'm not smart enough to figure it out…

Reply:
Holdover of old code